src/code_080023A4.c

```diff
@@ -1,12 +1,170 @@
+{
+    struct Unk_02023530 *var0 = gUnk_02023530 + playerId;
+
+    return gUnk_02024ED0[var0->unk65E][
+        var0->roomWidth / 16 * y + x
+    ];
+}
+
+u8 sub_08002434(struct Unk_02023530 *arg0, u16 x, u16 y)
+{
+    return gUnk_02024ED0[arg0->unk65E][
+        arg0->roomWidth / 16 * y + x
+    ];
+}
+
+u8 sub_08002470(u8 playerId, u16 x, u16 y)
+{
+    struct Unk_02023530 *var0 = gUnk_02023530 + playerId;
+
+    u16 roomWidth = var0->roomWidth / 16;
+    u16 roomHeight = var0->roomHeight / 16;
+
+    if (x * 0x10000 < 0) {
+        x = 0;
+    }
+    else if (roomWidth <= x) {
+        x = roomWidth - 1;
+    }
+
+    if (y * 0x10000 < 0) {
+        y = 0;
+    }
+    else if (roomHeight <= y) {
+        y = roomHeight - 1;
+    }
+
+    return gUnk_02024ED0[var0->unk65E][y * roomWidth + x];
+}
+
+u8 sub_080024F0(struct Unk_02023530 *arg0, u16 x, u16 y)
+{
+    u16 roomWidth = arg0->roomWidth / 16;
+    u16 roomHeight = arg0->roomHeight / 16;
+
+    if (x * 0x10000 < 0) {
+        x = 0;
+    }
+    else if (roomWidth <= x) {
+        x = roomWidth - 1;
+    }
+
+    if (y * 0x10000 < 0) {
+        y = 0;
+    }
+    else if (roomHeight <= y) {
+        y = roomHeight - 1;
+    }
+
+    return gUnk_02024ED0[arg0->unk65E][y * roomWidth + x];
+}
+
+u8 sub_0800255C(u8 playerId, u16 x, u16 y, u8 value)
+{
+    struct Unk_02023530 *var0 = gUnk_02023530 + playerId;
+    u32 var1 = var0->roomWidth / 16 * y + x;
+
+    u8 old = gUnk_02024ED0[var0->unk65E][var1];
+    gUnk_02024ED0[var0->unk65E][var1] = value;
+
+    return old;
+}
+
+struct Unk_0888562C *sub_080025AC(u8 playerId, u8 x, u8 y)
+{
+    struct Unk_02023530 *var0 = gUnk_02023530 + playerId;
+    struct Unk_0888562C *var1 = var0->unk1E0;
+    u8 var2 = gUnk_02024ED0[var0->unk65E][y * (var0->roomWidth / 16) + x];
+
+    for (;;) {
+        if (
+            var1->unk01 == var2 &&
+            var1->unk02 == x &&
+            var1->unk03 == y
+        ) {
+            return var1;
+        }
+
+        var1 = (void *) ((u32) var1 + var1->unk04);
+    }
+}
+
+struct Unk_0888562C *sub_08002624(u8 playerId, u8 x, u8 y, u8 arg3)
+{
+    struct Unk_02023530 *var0 = gUnk_02023530 + playerId;
+    struct Unk_0888562C *var1 = var0->unk1E0;
+    u8 var2 = gUnk_02024ED0[var0->unk65E][y * (var0->roomWidth / 16) + x];
+
+    for (;;) {
+        if (
+            var1->unk01 == var2 &&
+            var1->unk02 == x &&
+            var1->unk03 == y &&
+            --arg3 == 0xFF
+        ) {
+            return var1;
+        }
+
+        var1 = (void *) ((u32) var1 + var1->unk04);
+    }
+}
+
+u8 sub_080026A8(u8 playerId, u16 x, u16 y)
+{
+    struct Unk_02023530 *var0 = gUnk_02023530 + playerId;
+
+    return gUnk_02026D60[var0->unk65E][
+        var0->roomWidth / 16 * y + x
+    ];
+}
+
+u8 sub_080026F8(u16 roomId)
+{
+    return (*gUnk_08D637AC[gUnk_089331AC[roomId].objectListIdx])[0].type;
+}
+
+u8 sub_0800271C(u16 roomId, u8 arg1)
+{
+    return *(u8 *) (
```

Comment:
Maybe unk04 is a union? 
Also I don't see the point in casting to `u32 *`, given it's a `u8` array

Reply:
The `u32 *` cast I think is there so that it'd output an `ldr` instead of an `ldrb`, not sure.

Reply:
oh yes. sorry i read it wrong